### DISCUSSION TITLE
19.0.0-rc.0

### DISCUSSION THREAD

**Discussion by jamesdaniels** [angular/angularfire#3579] [General]
## What's Changed
* Allow use in Zoneless projects
* Depends on Angular v19, Firebase v11
* Now bundling schematics, this allows us to clean up some dependencies
* Drop locationId from the downloaded config on `ng add`, fixes #3556

## What's left
Before we release v19 final we have a couple things to fix up

* Make sure Zoneless is working well with Angular SSR, better leverage PendingTasks
* Dust off the rxfire dependency, it's a little long in the tooth
* We marked a couple flaky FirestoreCollectionGroup tests as pending, we should sort those
* Codesplit the now bundled schematics, will shave ~2mb off the tarball
* Update samples and docs to v19
* Address any show stoppers encountered while testing

## New Contributors
* @mmalerba made their first contribution in https://github.com/angular/angularfire/pull/3543
* @DellaBitta made their first contribution in https://github.com/angular/angularfire/pull/3570
* @hsubox76 made their first contribution in https://github.com/angular/angularfire/pull/3577

**Full Changelog**: https://github.com/angular/angularfire/compare/18.0.1...19.0.0-rc.0

<hr /><em>This discussion was created from the release <a href='https://github.com/angular/angularfire/releases/tag/19.0.0-rc.0'>19.0.0-rc.0</a>.</em>

**Comment by Pianisimo** [comment]
Right as it is it is not working out of the box and you get:
`error FirebaseError: Type does not match the expected instance. Did you pass a reference from a different Firestore SDK?`
The problem fixes if you manually add `"firebase": "^11.0.2"` to your deps.

**Reply by jamesdaniels** [reply]
Addressed in [19.0.0-rc.1](https://github.com/angular/angularfire/releases/tag/19.0.0-rc.1)

**Comment by metalllus** [comment]
Sorry to put this in here but I do not have the permission to create new discussions. Can someone please direct me to an up to date article/doc with the explanation how to query Firestore when the page is rendered on the server with SSR? I am just trying out the new angular 19 release and still if I use @angular/fire I do not get the data in the template that are queried with @angular/fire from Firestore. When I open page source I see my page data but the part that is supposed to show the data originating from Firestore has only a placeholder  tag in the source code. It is clear that the data is not there on the initial page load, only afterwards when the javascript kicks in on the client it is rendered. I would much appreciate a guideline how this is supposed to be done in Angular 19. Thank you very much.

**Reply by jamesdaniels** [reply]
Give [19.0.0-rc.1](https://github.com/angular/angularfire/releases/tag/19.0.0-rc.1) a spin once the build is on NPM, this should allow proper render blocking in Zoneless SSR applications.

**Reply by jamesdaniels** [reply]
In an upcoming release we'll have an improved API for "resuming" the snapshot listener from the server fetch.

**Comment by Soolan** [comment]
Hi devs,
1. Thank you for your efforts to release v19
2. Any ETA for the final release date?
